appserver/deployment/dol/src/main/java/com/sun/enterprise/deployment/util/ComponentValidator.java

```diff
@@ -468,70 +468,65 @@ protected void accept(EjbReference ejbRef) {
-                    // arises for remote @EJB annotations, so don't log it as warning.
-                    DOLUtils.getDefaultLogger().fine("Unresolved <ejb-link>: "+linkName);
-                } else {
-                    DOLUtils.getDefaultLogger().warning("Unresolved <ejb-link>: "+linkName);
-                }
-                return;
+                DOLUtils.getDefaultLogger().warning("Unresolved <ejb-link>: "+linkName);
             }
-        } else {
-
-            if( ejbRef.isEJB30ClientView() ) {
-
-                BundleDescriptor referringBundle = 
-                    ejbRef.getReferringBundleDescriptor();
-
-                // If we can verify that the current ejb 3.0 reference is defined
-                // in any Application Client module or in a stand-alone web module
-                // it must be remote business.
-                if( ( (referringBundle == null) && (getEjbBundleDescriptor() == null) )
-                    ||
-                    ((referringBundle != null) && (referringBundle.getModuleType() == DOLUtils.carType()))
-                    ||
-                    ( (getApplication() == null) &&
-                      (referringBundle.getModuleType() != null && referringBundle.getModuleType().equals(DOLUtils.warType())) ) ) {
-
-                    ejbRef.setLocal(false);
-
-                    // Double-check that target has a remote business interface of this
-                    // type.  This will handle the common error case that the target 
-                    // EJB has intended to support a remote business interface but
-                    // has not used @Remote to specify it, in which case
-                    // the interface was assigned the default of local business.
-
-                    if( !ejbReferee.getRemoteBusinessClassNames().contains
-                        (intfClassName) ) {
-                        String msg = "Target ejb " + ejbReferee.getName() + " for " +
-                            " remote ejb 3.0 reference " + ejbRef.getName() + 
-                            " does not expose a remote business interface of type " +
-                            intfClassName;
-                        throw new RuntimeException(msg);
-                    }
+            return;
+        }
+        if( ejbRef.isEJB30ClientView() ) {
+
+            BundleDescriptor referringBundle = 
+                ejbRef.getReferringBundleDescriptor();
+
+            // If we can verify that the current ejb 3.0 reference is defined
+            // in any Application Client module or in a stand-alone web module
+            // it must be remote business.
+            if (((referringBundle == null) && (getEjbBundleDescriptor() == null))
+                    || ((referringBundle != null) && (referringBundle.getModuleType() == DOLUtils.carType()))
+                    || ((getApplication() == null) && (referringBundle != null)
+                            && referringBundle.getModuleType() != null
```

Comment:
The above line is the fix for a potential NPE testing the condition. I assume this was simply a miss.
Rest of the change is caused by removing unnecessary `else` clauses. Behaviour change: NPE during condition check now does no longer occur.